plugins/meson/meson.mk

```diff
@@ -46,7 +46,26 @@ endef
 
 define $(PKG)_BUILD_$(BUILD)
     cd '$(SOURCE_DIR)' && python3 setup.py install \
-        --prefix='$(PREFIX)/$(TARGET)' && \
-    $(SED) -i 's,^#!/usr/bin/python3,#!/usr/bin/env python3,' \
-        '$(PREFIX)/$(TARGET)/bin/meson'
+        --prefix='$(PREFIX)/$(TARGET)'
+
+    # Awful hacks: we must hijack the python entry points here to install our
+    # site-packages path. This is because Meson is going to put the path to the
+    # real python interpreter and script it's been invoked with into the build
+    # rules file, bypassing any of our shell wrappers. This causes automatic
+    # reconfiguration to fail.
```

Comment:
What happens if you use`#!/$(PREFIX)/$(BUILD)/bin/python3` in the meson wrapper? And/or copy the python executable instead of the wrapper setting `PYTHONPATH`?

Reply:
The shebang line doesn't matter, Meson always puts the path to the real python executable into `build.ninja`. Copying the python3 executable also does not accomplish anything (and it's an even nastier hack in my opinion). While it would cause it to put "our" python3 into the generated build rules, `sys.path` would still not be set up correctly, and import of `mesonbuild` would fail unless it's available on the host system... which defeats the whole point of packaging meson in the first place. And yes, I have tried copying it. `sys.path` doesn't change.

The only way I see to fix this properly without patching Meson is to also package python3 properly. But even then we might still need some PATH hack to ensure that Meson finds the correct `ninja` executable.

Alternatively, we can just give up on this and go with my original thin wrapper for the native Meson.

This kind of set up with native python interpreted + bundled python packages is kind of volatile, anyway. Everything will probably fall apart if the user upgrades their python version, for example (without also rebuilding the relevant MXE packages).

Reply:
@tonytheodore so, what do you think about this?

Reply:
You're right that anyone using this plugin will already have `meson` and `ninja` installed, so a thin wrapper is probably the best option for now.

I had a quick look at building the `meson` branch of `taisei` and it looks like a very nice build system.

P.S. the [venv](https://docs.python.org/3/library/venv.html#module-venv) python module works better than setting PYTHONPATH for this, but doesn't help with the `meson` paths.

Reply:
@tonytheodore Alright, I'm going to reopen #1995 then. Please tell me if there are any issues remaining with it.

I also thought of trying to make use of venv, but unfortunately, the way it works requires setting up some environment variables in its `bin/activate` shell script, so that's not very useful here either. Maybe I should try contributing a way to override the `meson` invocation command upstream, so that we (and whoever else) can have a clean wrapper without patches.

Reply:
venv works fine without activate, [from docs](https://docs.python.org/3/library/venv.html#venv-def):
> There should be no need in other circumstances to activate a virtual environment

Meson's [invocation of itself](https://github.com/mesonbuild/meson/blob/master/mesonbuild/environment.py#L327-L331) seems to change a lot since the last release so a solution upstream would be great.